Fix and test log.info calls run for each migration

danjm opened the pull request:
In migrator/index.js, log.info is called before an after each migration. These calls are intended to produce breadcrumbs to be captured by sentry in cases where errors happen during or shortly after migrations are run. These calls were not causing any output to the console because the log.setLevel calls in ui/index.js were setting a 'warn value in local storage that was being used by logLevel in the background.

This commit fixes the problem by setting the `persist` param of setLevel to false, so that the background no longer reads the ui's log level.

Tests are added to verify that these logs are captured in sentry breadcrumbs when there is a migration error due to an invariant state.

## Manual Testing Steps

1. Checkout this branch and run `yarn dist`
2. Install and onboard, opting into metametrics while onboarding
3. Run `chrome.storage.local.get(({ data, meta }) => chrome.storage.local.set({ data: { ...data, PreferencesController: 5 }, meta: {...meta, version: 75} }, () => { window.location.reload() }))` in the background dev console
4. You should see logs like "Running migration 75" and "Migration 75 complete" for each of the migrations from 75 to 94.
5. There should be a network request to sentry, with breadcrumbs for the console log entries from 75 to 82.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

danjm committed:
Fix and test log.info calls run for each migration

In migrator/index.js, log.info is called before an after each migration.
These calls are intended to produce breadcrumbs to be captured by sentry
in cases where errors happen during or shortly after migrations are run.
These calls were not causing any output to the console because the log.setLevel
calls in ui/index.js were setting a 'warn value in local storage that was being
used by logLevel in the background.

This commit fixes the problem by setting the `persist` param of setLevel to
false, so that the background no longer reads the ui's log level.

Tests are added to verify that these logs are captured in sentry breadcrumbs
when there is a migration error due to an invariant state.

danjm commented on app/scripts/background.js:
```diff
@@ -88,7 +88,7 @@ const metamaskInternalProcessHash = {
 
 const metamaskBlockedPorts = ['trezor-connect'];
 
-log.setDefaultLevel(process.env.METAMASK_DEBUG ? 'debug' : 'info');
```
whoops... I committed the wrong stash here...

danjm replied:
fixed

Gudahtt reviewed (dismissed):
LGTM!

Gudahtt commented on test/e2e/tests/errors.spec.js:
```diff
@@ -307,6 +319,51 @@ describe('Sentry errors', function () {
+          const lastMigrationLog =
+            migrationLogBreadcrumbs[migrationLogBreadcrumbs.length - 1];
+
+          assert.equal(migrationLogBreadcrumbs.length, 8);
```
Chrome e2e tests failing on this line

Gudahtt replied:
Testing shows that the logs are showing up correctly, and the breadcrumbs are captured correctly. The logs are just delayed. They aren't showing up in time for this error report.

Gudahtt replied:
Perhaps we can test this using a post-initialization failure instead? I know we envisioned this being useful for initialization failures, but... maybe there's no way for us to guarantee they show up for those.

danjm replied:
@Gudahtt How did you test it to see the breadcrumbs are captured correctly?

danjm replied:
I am pretty sure I did successfully run these e2e tests... but maybe the branch or build wasn't in the exact same state that the branch is now...

Gudahtt replied:
I used `console.log` to print the breadcrumbs in the test, and they are correct for the point-in-time when the error occurred in the browser console (which I checked via using `--leave-running`). The logs you're trying to capture here are just happening later.

danjm committed:
Improve breadcrumb message matching

The test modified in this commit asserts eqaulity of  messages from breadcrumbs
and hard coded expected results. This could cause failures, as sometimes the
messages contain whitespace characters. This commit ensures the assertions only
check that the expected string is within the message string, ignoring extra
characters.

Gudahtt reviewed (approved):
LGTM!
